### PR TITLE
Redundant native cuda code removed

### PR DESCRIPTION
**Pull request by FelixTUD**
The redundant native cuda code has been removed, leaving only alpaka as an option for the accelerator.

Cuda is still supported as an accelerator option by alpaka.

Better readability of the code and simpler maintenance.

### CODE REVIEW

File: example/CMakeLists.txt [hunk 1]
```diff
@@ -37,18 +37,7 @@ if(${ISAAC_STEREO} STREQUAL "Anaglyph")
...
-	REMOVE_DEFINITIONS(REMOVE_ITEM "-DISAAC_ALPAKA")
-	set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS}" -std=c++11)
-    list(REMOVE_DUPLICATES CUDA_NVCC_FLAGS)
-	SET_SOURCE_FILES_PROPERTIES(${CMAKE_SOURCE_DIR}/example.cpp PROPERTIES CUDA_SOURCE_PROPERTY_FORMAT OBJ)
-	CUDA_ADD_EXECUTABLE(isaac_example_cuda ${CMAKE_SOURCE_DIR}/example.cpp)
-	target_link_libraries(isaac_example_cuda ${ISAAC_LIBRARIES})
-	install(TARGETS isaac_example_cuda DESTINATION bin )
-endif()
+ALPAKA_ADD_EXECUTABLE(isaac_example ${CMAKE_SOURCE_DIR}/example.cpp)
+set(ISAAC_LIBRARIES ${ISAAC_LIBRARIES} "alpaka::alpaka")
```
Comment (PrometheusPi) [inline example/CMakeLists.txt:41]: at @psychocoderHPC: 
@FelixTUD mentioned that here alpaka probably changed with the last version tested - can you confirm this?

File: lib/ISAACConfig.cmake [hunk 2]
```diff
@@ -161,34 +152,20 @@ set(ISAAC_DEFINITIONS ${ISAAC_DEFINITIONS} -DBOOST_ALL_NO_LIB)
...
-    endif()
+find_package(alpaka QUIET)
+if (!alpaka_FOUND)
+    set(ISAAC_PRIVATE_FOUND false)
+    set(ISAAC_DEPENDENCY_HINTS ${ISAAC_DEPENDENCY_HINTS} "\n--   Cuda or Alpaka")
+else()
+    set(ISAAC_INCLUDE_DIRS ${ISAAC_INCLUDE_DIRS} ${alpaka_INCLUDE_DIRS})
+    set(ISAAC_LIBRARIES ${ISAAC_LIBRARIES} ${alpaka_LIBRARIES})
+    set(ISAAC_DEFINITIONS ${ISAAC_DEFINITIONS} ${alpaka_DEFINITIONS})
+    set(ISAAC_DEFINITIONS ${ISAAC_DEFINITIONS} ${ALPAKA_DEV_COMPILE_OPTIONS})
```
Comment (PrometheusPi) [inline lib/ISAACConfig.cmake:167]: at @psychocoderHPC :
Are these still step by alpaka. 
(@FelixTUD Please correct my question if it is too imprecise.)

File: example/CMakeLists.txt (hunk 1 above)
Comment (psychocoderHPC) [reply example/CMakeLists.txt:41]: This look valid and should work. @FelixTUD do you tested it with PIConGPU dev?

File: lib/ISAACConfig.cmake [hunk 3]
```diff
@@ -161,34 +152,20 @@ set(ISAAC_DEFINITIONS ${ISAAC_DEFINITIONS} -DBOOST_ALL_NO_LIB)
...
-        set(ISAAC_DEPENDENCY_HINTS ${ISAAC_DEPENDENCY_HINTS} "\n--   Cuda or Alpaka")
-    else()
-        set(ISAAC_INCLUDE_DIRS ${ISAAC_INCLUDE_DIRS} ${alpaka_INCLUDE_DIRS})
-        set(ISAAC_LIBRARIES ${ISAAC_LIBRARIES} ${alpaka_LIBRARIES})
-        set(ISAAC_DEFINITIONS ${ISAAC_DEFINITIONS} ${alpaka_DEFINITIONS})
-        set(ISAAC_DEFINITIONS ${ISAAC_DEFINITIONS} ${ALPAKA_DEV_COMPILE_OPTIONS})
-        set(ISAAC_DEFINITIONS ${ISAAC_DEFINITIONS} "-DISAAC_ALPAKA")
-    endif()
+find_package(alpaka QUIET)
+if (!alpaka_FOUND)
```
Comment (psychocoderHPC) [inline lib/ISAACConfig.cmake:160]: CMAKE NOT is `if(NOT alpaka_FOUND)`

File: lib/ISAACConfig.cmake (hunk 2 above)
Comment (psychocoderHPC) [reply lib/ISAACConfig.cmake:167]: IMO this is not available anymore. The problem is that ISAAC is not providing a target.
ISAAC needs to provide an `INTERFACE` target and then you can use `target_link_libraries`

**Review by psychocoderHPC — CHANGES_REQUESTED** [review]

**Comment by psychocoderHPC** [comment]
@FelixTUD IF you do not know whats to do we should do a VC next week.

File: example/CMakeLists.txt (hunk 1 above)
Comment (FelixTUD) [reply example/CMakeLists.txt:41]: There is no Problem with PIConGPU dev, only with the standalone example, currently I added it only to the example and I think it should replace the 4 alpaka defines discussed below in the lib/ISAACConfig.cmake, right?

**Comment by FelixTUD** [comment]
@psychocoderHPC I think that would be better, right know I only removed all the cuda dependent code and left the rest as is, but the cmake files need a general overhaul

**Comment by FelixTUD** [comment]
@psychocoderHPC The requested changes are squashed into the commit

**Review by psychocoderHPC — APPROVED** [review]

**Pull request merged**